options/posix/generic/pwd-stubs.cpp

```diff
@@ -8,16 +8,9 @@
 #include <mlibc/debug.hpp>
 
 namespace {
-	thread_local passwd global_entry;
+	FILE *global_file; // Used by setpwent/getpwent/endpwent.
```

Comment:
It's probably nicer to have this as a static local variable in a wrapper function. Or at least have some ``get_global_file()`` static function that does the opening of the file because right now that code is duplicated 3x. Then also make a function like ``close_global_file()`` for when it needs to be closed.